src/monodroid/jni/monodroid-glue.c

```diff
@@ -704,55 +723,104 @@ typedef void* (*mono_mkbundle_init_ptr) (void (*)(const MonoBundledAssembly **),
 mono_mkbundle_init_ptr mono_mkbundle_init;
 void (*mono_mkbundle_initialize_mono_api) (const BundleMonoAPI *info);
 
-static void
-setup_bundled_app (const char *libappso)
+static char*
+get_full_dso_path (const char *base_dir, const char *dso_path, mono_bool *needs_free)
 {
-	void *libapp;
+	if (needs_free != NULL)
+		*needs_free = FALSE;
+	if (!dso_path)
+		return NULL;
 
-	libapp = dlopen (libappso, RTLD_LAZY);
+	if (base_dir == NULL || dso_path [0] == MONODROID_PATH_SEPARATOR_CHAR)
```

Comment:
(Since I haven't tried to check...)

Can `get_full_dso_path()` be called from the Windows/Designer codepaths?  If so, this could be a Windows path.  Perhaps we need an `is_path_rooted(const char*)` function?

Reply:
Yeah, this is just a quick check "for now", to get the code working on device asap. I'll add the function to TODO